File: baekjoon-typescript/2908.ts
import {readFileSync} from "fs";

const fileName = __filename.split('\\').pop().split('.')[0];
const path = process.platform === "linux" ? "/dev/stdin" : `./input/${fileName}.txt`;
const args = readFileSync(path).toString().trim().split('\n') || [];

function main(args) {
    const arr: string[] = String(args[0]).split(" ");

    const rs: number = Math.max(
        reverseAndConvertNumber(String(arr[0]).split("")),
        reverseAndConvertNumber(String(arr[1]).split("")),
    )

    console.log(rs);
}

main(args);

function reverseAndConvertNumber(arr: string[]): number {
    let str = ""

    for (let i = arr.length - 1; i >= 0; i--) {
        str += arr[i]
    }

    return Number(str)
}

File: baekjoon-go/2908.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func main() {

	var a, b string
	fmt.Scanf("%s %s", &a, &b)

	rs := math.Max(
		reverseAndConvertNumber([]rune(a)),
		reverseAndConvertNumber([]rune(b)),
	)
	fmt.Println(rs)
}

func reverseAndConvertNumber(arr []rune) float64 {
	var str string

	for i := len(arr) - 1; i >= 0; i-- {
		str += string(arr[i])
	}

	rs, _ := strconv.Atoi(str)

	return float64(rs)
}
